hbase-server/src/main/java/org/apache/hadoop/hbase/ipc/RpcServer.java

```diff
@@ -1254,8 +1255,10 @@ private boolean processAllResponses(final Connection connection) throws IOExcept
           if (call == null) {
             return true;
           }
+          metrics.removeCallFromResponseQueue(call.response.getRemaining());
           if (!processResponse(call)) {
             connection.responseQueue.addFirst(call);
+            metrics.addCallToResponseQueue(call.response.getRemaining());
```

Comment:
Isn't the right way to do this something like..

```
 before = call.response.getRemaining();  <=== 1
  if (!processResponse(call)) {
   after = call.response.getRemaining();
   metrics.decr(after-before);  <=== 2
} else {
   metrics.decr(before);
}

```
The difference is you are not decrementing until the bytes are actually flushed, otherwise if a metrics poller polls between (1) and (2), there is some weird state in between, perhaps ^^ is more desirable for you.

Reply:
I got you.